scrapy/core/downloader/handlers/ftp.py

```diff
@@ -63,22 +63,22 @@ def close(self):
 _CODE_RE = re.compile(r"\d+")
 
 
-class FTPDownloadHandler(object):
+class FTPDownloadHandler:
     lazy = False
 
     CODE_MAPPING = {
         "550": 404,
         "default": 503,
     }
 
-    def __init__(self, crawler):
-        self.default_user = crawler.settings['FTP_USER']
-        self.default_password = crawler.settings['FTP_PASSWORD']
-        self.passive_mode = crawler.settings['FTP_PASSIVE_MODE']
+    def __init__(self, settings, crawler=None):
```

Comment:
crawler is not needed here

Reply:
It is not a goal for `__init__` methods of all downloader handlers to have the same signature, because downloader handlers are created using `create_instance` - it gives us flexibility to use any parameters which make sense in `__init__`, while still being able to create them in a generic way.